src/Nethermind/Nethermind.Blockchain/Headers/IHeaderStore.cs

```diff
@@ -14,6 +14,7 @@ public interface IHeaderStore
     void BulkInsert(IReadOnlyList<BlockHeader> headers);
     BlockHeader? Get(Hash256 blockHash, bool shouldCache, long? blockNumber = null);
     void Cache(BlockHeader header);
+    void Cache(Hash256 blockHash, long blockNumber);
```

Comment:
I don't like adding another method. IBlockTree is too fat.

Reply:
Ok, trying to find out why blocks don't cache properly; that might resolve it